primitives/phragmen/src/helpers.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019-2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Helper methods for phragmen.
+
+use crate::{Assignment, ExtendedBalance, IdentifierT, StakedAssignment};
+use sp_runtime::PerThing;
+use sp_std::prelude::*;
+
+/// Converts a vector of ratio assignments into ones with absolute budget value.
+pub fn assignment_ratio_to_staked<A: IdentifierT, T: PerThing, FS>(
+	ratio: Vec<Assignment<A, T>>,
+	stake_of: FS,
+) -> Vec<StakedAssignment<A>>
+where
+	for<'r> FS: Fn(&'r A) -> ExtendedBalance,
+	T: sp_std::ops::Mul<ExtendedBalance, Output = ExtendedBalance>,
+	ExtendedBalance: From<<T as PerThing>::Inner>,
+{
+	ratio
+		.into_iter()
+		.map(|a| {
+			let who = a.who.clone();
+			a.into_staked(stake_of(&who), true)
```

Comment:
```suggestion
			a.into_staked(stake_of(&stake), true)
```

Reply:
Is it different though? both need one temp useless allocation since a is being moved in `into_staked `